### ISSUE TITLE
Starting STK without menu gives bad performance

### ISSUE THREAD

**Issue by Arthur-D** [supertuxkart/stk-code#2355]
Whenever I start STK with -N or -R flags performance goes down a lot. If I set up a new race, the problem goes away. Just restarting the race does not help. I would really like to know if anyone can reproduce this issue. I'm using the proprietary NVIDIA driver under Linux.

**Comment by MiroslavR** [comment]
I cannot reproduce this with either of these options. I'm also using the proprietary NVIDIA driver on Linux (version 346.72).

**Comment by Elderme** [comment]
I'm also using proprietary NVIDIA driver on Linux, and I cannot reproduce.

Arthur-D, what is your OpenGL version and your graphic card? This performance issue may depend on available OpenGL extensions. There are at least three ways of rendering the scene depending on the available OpenGL extensions: one way using only OpenGL3.2 functions, one way with OpenGL4.x functions, and one way with OpenGL4.x functions and "AZDO" extensions.

**Comment by Arthur-D** [comment]
I have a Zotac NVIDIA GTS 250 1GB version with version 340.93 of the driver. Specifically, it only supports up to OpenGL 3.3.

**Comment by VOTProductions** [comment]
Cannot confirm on Mesa, AMD Radeon HD7750.

**Comment by deveee** [comment]
The same bug has been reported here:
https://github.com/supertuxkart/stk-code/issues/1479
even with stdout.log:
http://pastebin.com/8Xu6VwuN

**Comment by Arthur-D** [comment]
Yes that's true. I still have this issue though I've not tried installing the very latest driver as of today. Makes it very annoying for making tracks or otherwise debugging since I need to go through the menus every time.

**Comment by deveee** [comment]
Is this CPU or GPU related? You can look at CPU usage. If it's much higher, then you could use sysprof. It should show which function has the highest CPU usage.

**Comment by Arthur-D** [comment]
It seems to be GPU related, see attached screenshots of profiler.

After going through the menus:
![cocoa_temple-2015 11 18_13 57 50](https://cloud.githubusercontent.com/assets/5515261/11241652/d2eafbc0-8dfc-11e5-916e-db7ed848a36a.png)
Using -R flag:
![cocoa_temple-2015 11 18_13 57 09](https://cloud.githubusercontent.com/assets/5515261/11241678/071cad76-8dfd-11e5-8f2b-607cbfe6ef82.png)

**Comment by auriamg** [comment]
sync stall means the game is waiting for the GPU to complete, can you please look at the GPU profiler too? The second profiler right under

**Comment by Arthur-D** [comment]
Menus:
![cocoa_temple-2015 11 19_21 54 16](https://cloud.githubusercontent.com/assets/5515261/11284196/c7d43384-8f08-11e5-808f-7a0251e5d082.png)
Direct (-R):
![cocoa_temple-2015 11 19_21 55 34](https://cloud.githubusercontent.com/assets/5515261/11284210/d4ca440c-8f08-11e5-87a9-0fde8784c5a6.png)

Not sure exactly what to look at - having all of those numbers shown at the same time would make it easier to compare I guess.

**Comment by auriamg** [comment]
There is no bar that is obviously much larger in the second compared to the first, so no obvious answer unfortunately. Does the graphics level make any difference?

**Comment by Arthur-D** [comment]
Yes, when I go to level 2, performance is as expected. As soon as I go to graphics level 3, I get 14 FPS as lowest, which is not normal at all. Going through menus the very lowest I see is 47 FPS, normally more.

**Comment by Arthur-D** [comment]
Profiler results from starting a race in Cocoa Temple with no AI, at graphics level 3.
In both cases, I just stood some seconds at the start line, I did not drive.

Going through menus, i.e. normal performance: https://www.dropbox.com/s/c6wrh5xhhi7aktq/profiling-goodfps.csv.zip?dl=0
Starting with -R, i.e. bad and unusual performance: https://www.dropbox.com/s/8961m1ulps9nrkr/profiling-badfps.csv.zip?dl=0

**Comment by Arthur-D** [comment]
[stdout.log-goodfps.txt](https://github.com/supertuxkart/stk-code/files/39610/stdout.log-goodfps.txt)
[stdout.log-badfps.txt](https://github.com/supertuxkart/stk-code/files/39611/stdout.log-badfps.txt)

Here's the stdout.log files from two different startups, again one with going through menus first and another with -R.

**Comment by deveee** [comment]
In your profiling-badfps.csv you can see very big values once per few updates. For example 12, 17, 22 line. Try to use sysprof because it shows also function names. It could also show if the graphics driver/openal/etc are busy. Example output:
![zrzut ekranu z 2015-11-22 16-03-04](https://cloud.githubusercontent.com/assets/5108558/11324486/3813471c-9133-11e5-8f5c-aa798a3156e2.png)

I see in stkAnon's stdout.log that he is running the game using geforce GTX 650, which is OpenGL 4.x. I have a bit worse, but quite similar graphics card (600 series) and it was always working for me without issues. So I suppose that the problem is somewhere else.

**Comment by Arthur-D** [comment]
I don't know if I did this correctly, or if I should have let it run longer, but here they are:
https://www.dropbox.com/s/8wqgnxun0a6ujmq/STK-goodfps?dl=0
https://www.dropbox.com/s/gj6d7aiy29nbohk/STK-badfps?dl=0

**Comment by Arthur-D** [comment]
Just reporting that I changed graphics card to a current generation NVIDIA, so I no longer have this issue. However I still think the bug is strange and annoying, but most people won't ever get this and I worked around it by getting newer hardware, so this could potentially be closed. At least I will change the milestone to Later, thus keeping it open in case someone else should get the same issue. Thanks for participating.
